platform/plugins/pid/PidService.ts

```diff
@@ -6,35 +6,34 @@ import { PidFile } from './PidFile';
 import { LoggerFactory } from '../../logger';
 
 export class PidService {
-  private readonly pid$: Observable<PidFile | void>;
+  private readonly pid$: Observable<undefined>;
   private subscription?: Subscription;
 
-  constructor(pidConfig$: Observable<PidConfig | void>, logger: LoggerFactory) {
+  constructor(
+    pidConfig$: Observable<PidConfig | undefined>,
+    logger: LoggerFactory
+  ) {
     this.pid$ = pidConfig$
-      .map(
-        config =>
-          config !== undefined
-            ? new PidFile(process.pid, config, logger)
-            : undefined
-      )
       // Explanation of `switchMap`:
       // It's kinda like a normal `flatMap`, except it's producing observables
       // and you _only_ care about the latest observable it produced. It's
       // usually used if you need to control what happens both when you create
       // and when you're done with an observable, like here where we want to
       // write the pid file we receive a pid config, and delete it when we
       // receive new config values (or when we stop the pid service).
-      .switchMap(pid => {
-        if (pid === undefined) {
-          // If pid is not specified, we return an observable that does nothing
-          return new Observable<PidFile | void>(noop);
+      .switchMap(config => {
+        if (config === undefined) {
+          // If there is no pid config we return an observable that does nothing
+          return new Observable(noop);
         }
 
         // Otherwise we return an observable that writes the pid when
         // subscribed to and deletes it when unsubscribed (e.g. if new config
```

Comment:
Yep, it's doing that now:

<img width="720" alt="screen shot 2017-07-07 at 18 09 41" src="https://user-images.githubusercontent.com/5259/27966437-8941b2ce-633f-11e7-8aec-a171bd1405aa.png">

I added some newlines here, and performed the sighup in another window. But just change the config, then sighup the process (`kill -1` it), and it should re-apply the config. I specified `logging.level: debug` first in the kibana config to see the debug logs.